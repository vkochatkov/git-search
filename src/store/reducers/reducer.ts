import {
  CLEAR_USERS,
  GET_REPOS,
  GET_USER,
  SEARCH_USERS,
  SET_LOADING,
} from '../actionTypes';
import { AnyAction } from 'redux';

export const initialState = {
  user: {},
  users: [],
  loading: false,
  repos: [],
};

export const reducer = (state = initialState, action: AnyAction) => {
  switch (action.type) {
    case SET_LOADING:
      return {
        ...state,
        loading: true,
      };
    case SEARCH_USERS:
      return {
        ...state,
        users: action.payload,
        loading: false,
      };
    case GET_USER:
      return { ...state, user: action.payload, loading: false };
    case GET_REPOS:
      return {
        ...state,
        repos: action.payload,
        loading: false,
      };
    case CLEAR_USERS:
      return {
        user: {},
        users: [],
        loading: false,
        repos: [],
      };
    default:
      return state;
  }
};
