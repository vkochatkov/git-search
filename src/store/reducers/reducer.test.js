// ./**tests**/reducer_test.js
import { initialState, reducer } from './reducer';

describe('user reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle SET_LOADING', () => {
    expect(
      reducer(initialState, {
        type: 'SET_LOADING',
      }),
    ).toMatchSnapshot();
  });
});
