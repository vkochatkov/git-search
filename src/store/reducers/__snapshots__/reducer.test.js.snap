// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user reducer should handle SET_LOADING 1`] = `
Object {
  "loading": true,
  "repos": Array [],
  "user": Object {},
  "users": Array [],
}
`;

exports[`user reducer should return the initial state 1`] = `
Object {
  "loading": false,
  "repos": Array [],
  "user": Object {},
  "users": Array [],
}
`;
